Firestore/Example/Tests/SpecTests/FSTMockDatastore.mm

```diff
@@ -290,11 +295,13 @@ + (instancetype)mockDatastoreWithWorkerDispatchQueue:(FSTDispatchQueue *)workerD
   static DatabaseInfo database_info{DatabaseId{"project", "database"}, "persistence", "host",
                                     false};
 
-  FSTEmptyCredentialsProvider *credentials = [[FSTEmptyCredentialsProvider alloc] init];
+  // Note that we purposely don't bother to cleanup the EmptyCredentialsProvider instances.
```

Comment:
This is fine by me.

Here's another alternative that I put together to demonstrate what I meant by (d) in the first comment: 
https://github.com/firebase/firebase-ios-sdk/commit/3020ae0255300bd55048c126a5e2bdfd3160b9e1

(See the rsgowman/port_auth_suggestion  branch).